crates/bevy_time/src/lib.rs

```diff
@@ -64,18 +79,26 @@ pub fn create_time_channels() -> (TimeSender, TimeReceiver) {
 /// there to this system through channels. Otherwise the time is updated in this system.
 fn time_system(
     mut time: ResMut<Time>,
+    update_strategy: Res<TimeUpdateStrategy>,
     time_recv: Option<Res<TimeReceiver>>,
-    mut has_received_time: Local<bool>,
 ) {
-    if let Some(time_recv) = time_recv {
+    let new_time = if let Some(time_recv) = time_recv {
         // TODO: Figure out how to handle this when using pipelined rendering.
         if let Ok(new_time) = time_recv.0.try_recv() {
-            time.update_with_instant(new_time);
-            *has_received_time = true;
-        } else if *has_received_time {
+            new_time
+        } else {
             warn!("time_system did not receive the time from the render world! Calculations depending on the time may be incorrect.");
+            Instant::now()
```

Comment:
Not sure if this matters, but this is a change in behavior from before. This is updating the time with Instant::now(), while the old code was skipping updating the time if nothing was received. I'm not sure if either way is more right or wrong as skipping an update could be just as bad as assuming Instant::now() depending on what caused the time not to be sent.

Reply:
I think we can keep the current behavior for now. With the way bevy is sending the time from render right now this shouldn't ever be hit. Issues would only arise when people are doing custom stuff. In those cases, I'm not sure what the better behavior would be. We can just punt that into the future when we have more concrete evidence of what is better. It'll be easy to change things to skip the update if this is found to be problematic.